pkg/controller/job/job_controller_test.go

```diff
@@ -705,7 +867,7 @@ func TestSyncJobPastDeadline(t *testing.T) {
 			}
 
 			// job & pods setup
-			job := newJob(tc.parallelism, tc.completions, tc.backoffLimit, batch.NonIndexedCompletion)
+			job := newJob(false, tc.parallelism, tc.completions, tc.backoffLimit, batch.NonIndexedCompletion)
```

Comment:
I said this early, and I don't know why it got lost.
Rather than adding more arguments, let's just modify the job after `newJob` for the tests that need it. Honestly, I should have done the same for completion mode.

Reply:
yes :100:

Reply:
I think we discussed this in the e2e test instead last time haha, done.